client/src/components/ChallengeCard.tsx

```diff
@@ -11,24 +10,22 @@ function ChallengeCard({
   id,
   difficulty,
 }: ChallengeDetailsShort) {
-
-  const getDifficultyStars = useCallback((difficulty: number) => {
-    return Array.from({ length: difficulty + 1 }, (_, i) => <StarFill key={i} />);
-  }, [difficulty]);
+  const getDifficultyStars = useCallback(
+    (difficulty: number) => {
+      return Array.from({ length: difficulty + 1 }, (_, i) => (
+        <StarFill key={i} />
+      ));
+    },
+    [difficulty],
```

Comment:
I don't think difficulty is a dependency here.

Reply:
That's true. This should be refactored to useMemo with difficulty as a dependency, or remove difficulty as an input.